ndcube/cube_utils.py

```diff
@@ -182,10 +178,13 @@ def index_sequence_as_cube(cubesequence, item):
     item_tuple = tuple(item_list)
     if item is None or (isinstance(item, tuple) and None in item):
         raise IndexError("None indices not supported")
-    try:
-        return get_cube_from_sequence(cubesequence, item_tuple, type_slice='sequence_as_cube')
-    except:
-        return get_cube_from_sequence(cubesequence, item_tuple)
+
+    # if the slicing is done across one cube. Then we can pass the item_tuple as
+    # (int, rest of the information) not (int, int , rest of the information)
+    if isinstance(item_tuple[0], int) and isinstance(item_tuple[1], int):
+        if item_tuple[0] == item_tuple[1]:
+            return get_cube_from_sequence(cubesequence, item_tuple[1::])
```

Comment:
As best as I can tell, ```item_tuple``` is either 
(sequence_index_or_slice, cube_index_or_slice_dim0, cube_index_or_slice_dim1, cube_index_or_slice_dim2, etc) 
OR
(sequence_index_or_slice, [first_cube_index_or_slice_dim0, last_cube_index_or_slice_dim0], cube_index_or_slice_dim1, cube_index_or_slice_dim2, etc.) 

Is this correct so far?

Assuming it is, I don't understand a few things about the above if statement:
* Why it must be the case that ```item_tuple[0] == item_tuple[1]```
* Why ```item_tuple[1]``` must also be an ```int``` for the return line below to be executed.  Isn't all that matters is the ```item_tuple[0]``` is an ```int```, i.e. on one cube is being sliced?  So couldn't ```item_tuple[1]``` also be a slice?